src/components/carousel/CarouselList.vue

```diff
@@ -114,160 +119,180 @@ export default {
+            this.hold = Date.now()
             this.dragging = true
-            this.dragStartX = event.touches ? event.touches[0].clientX : event.clientX
-            window.addEventListener(event.touches ? 'touchmove' : 'mousemove', this.dragMove)
-            window.addEventListener(event.touches ? 'touchend' : 'mouseup', this.dragEnd)
+            this.touch = !!event.touches
+            this.dragStartX = this.touch ? event.touches[0].clientX : event.clientX
+            window.addEventListener(this.touch ? 'touchmove' : 'mousemove', this.dragMove)
+            window.addEventListener(this.touch ? 'touchend' : 'mouseup', this.dragEnd)
         },
         dragMove(event) {
-            this.dragEndX = event.touches ? event.touches[0].clientX : event.clientX
-            const deltaX = this.dragEndX - this.dragStartX
-            this.delta = deltaX < 0 ? Math.abs(deltaX) : -Math.abs(deltaX)
+            if (!this.dragging) return
+            const dragEndX = event.touches ? event.touches[0].clientX : event.clientX
+            this.delta = this.dragStartX - dragEndX
             if (!event.touches) {
                 event.preventDefault()
             }
         },
-        dragEnd(event) {
-            const signCheck = 1 * sign(this.delta)
-            const results = Math.round(Math.abs(this.delta / this.itemWidth) + 0.15)// Hack
-            this.switchTo(this.activeItem + signCheck * results)
+        dragEnd() {
+            if (!this.dragging && !this.hold) return
+            if (this.hold) {
+                const signCheck = sign(this.delta)
+                const results = Math.round(Math.abs(this.delta / this.itemWidth) + 0.15)// Hack
+                this.switchTo(this.scrollIndex + signCheck * results)
+            }
             this.dragging = false
             this.delta = 0
-            window.removeEventListener(event.touches ? 'touchmove' : 'mousemove', this.dragMove)
-            window.removeEventListener(event.touches ? 'touchend' : 'mouseup', this.dragEnd)
-        }
-    },
-    created() {
-        this.initConfig()
-        if (typeof window !== 'undefined') {
-            window.addEventListener('resize', this.update)
+            window.removeEventListener(this.touch ? 'touchmove' : 'mousemove', this.dragMove)
+            window.removeEventListener(this.touch ? 'touchend' : 'mouseup', this.dragEnd)
+        },
+        refresh() {
+            this.$nextTick(() => {
+                this.refresh_++
+            })
         }
     },
     mounted() {
-        this.$nextTick(() => {
-            this.update()
-        })
+        this.observer = new ResizeObserver(this.refresh)
+        this.observer.observe(this.$el)
+        window.addEventListener('resize', this.resized)
+        this.resized()
+        if (this.$attrs.config) {
+            throw new Error('The config prop was removed, you need to use v-bind instead')
+        }
```

Comment:
I think you can remove this

Reply:
This is important when releasing breaking changes, not everybody reads the changelog and if you don't trigger an error people may not notice. It should be removed on the following major